# **Palmer Penguins**
### Author: *Paul Cahill*

![Penguins](https://upload.wikimedia.org/wikipedia/commons/thumb/a/a3/Gentoo_Penguin_AdF.jpg/640px-Gentoo_Penguin_AdF.png)

This notebook contains my analysis of the famous palmer penguins dataset.

The dataset is available on GitHub [here](https://allisonhorst.github.io/palmerpenguins/).

***
## Importing the Data

Pandas is used for the DataFrame data structure.

It allows us to investigate CSV files, amongst other features.

In [6]:
#Data frames.
import pandas as pd

***
## Loading the Data

The palmer penguins dataset is loaded from a URL.

In [7]:
#Load the Penguins data set.
df = pd.read_csv("https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv")

Having loaded the data, it can now be examined on a surface level.


In [8]:
#Looking
df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
339  Gentoo     Biscoe             NaN            NaN                NaN   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  
0         3750.0    MALE  
1         3800.0  FEMALE  
2         3250.0  FEMALE  
3            NaN     NaN  
4         3450.0  FEMALE  
..           ...     ...  
339          NaN     NaN  
340       4850.0  FEMALE  
341       5750.0    MALE  
342       5200.0  FEMALE  
343       5400.0    MALE  

[344 rows x 7 columns]

In [9]:
# Grouping by species and island and counting the number of penguins
penguins_by_species_island = df.groupby(['species', 'island']).size().reset_index(name='count')

# Displaying the counts
penguins_by_species_island

species     island  count
0     Adelie     Biscoe     44
1     Adelie      Dream     56
2     Adelie  Torgersen     52
3  Chinstrap      Dream     68
4     Gentoo     Biscoe    124

There are three species of penguins included in this study:

- 68 **Chinstrap** penguins were recorded, each on Dream island.
- 124 **Gentoo** penguins were recorded, each on Biscoe.
- 152 **Adelie** penguins were recorded, with the species being present on both Dream and Biscoe, plus Torgersen.

***
## Analysing The Data

Having an understanding of the structure of the data, it can now be analysed with greater depth.




***

### End